discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala

```diff
@@ -99,16 +103,28 @@ class KubernetesApiSimpleServiceDiscovery(system: ActorSystem) extends SimpleSer
       entity <- response.entity.toStrict(resolveTimeout)
 
       podList <- {
-        log.debug("Kubernetes API entity: [{}]", entity.data.utf8String)
-
-        val unmarshalled = Unmarshal(entity).to[PodList]
 
-        unmarshalled.failed.foreach { t =>
-          log.error(t, "Failed to unmarshal Kubernetes API response status [{}]; check RBAC settings",
-            response.status.value)
+        response.status match {
+          case StatusCodes.OK =>
+            log.debug("Kubernetes API entity: [{}]", entity.data.utf8String)
+            val unmarshalled = Unmarshal(entity).to[PodList]
+            unmarshalled.failed.foreach { t =>
+              log.error(t, "Failed to unmarshal Kubernetes API response.  Status code: [{}]; Response body: [{}]",
+                response.status.value, entity)
```

Comment:
does `entity.toString` contain the whole response or is it truncated?

Reply:
It'll truncate. Will switch to Unmarshal as a String.